core/upgrade/upgrade.go

```diff
@@ -3,22 +3,87 @@
 
 package upgrade
 
-import "time"
+import "github.com/juju/errors"
+
+var ErrAlreadyAtState = errors.ConstError("already at state")
+
+// State represents the state of the upgrade.
+type State int
+
+const (
+	Created State = iota
+	Started
+	DBCompleted
+	StepsCompleted
```

Comment:
Should we include an aborted state like for migrations? Even if we don't use it programmatically to begin with, it's an option for manual intervention.